digdag-standards/src/main/java/io/digdag/standards/operator/param/ParamSetOperatorFactory.java

```diff
@@ -0,0 +1,79 @@
+package io.digdag.standards.operator.param;
+
+import com.google.common.base.Optional;
+import com.google.inject.Inject;
+import io.digdag.client.config.Config;
+import io.digdag.client.config.ConfigException;
+import io.digdag.spi.Operator;
+import io.digdag.spi.OperatorContext;
+import io.digdag.spi.OperatorFactory;
+import io.digdag.spi.TaskResult;
+import io.digdag.util.BaseOperator;
+
+import java.util.List;
+
+public class ParamSetOperatorFactory
+        implements OperatorFactory
+{
+    private final Config systemConfig;
+    private final ParamServerClient paramServerClient;
+
+    @Inject
+    public ParamSetOperatorFactory(Config systemConfig, ParamServerClient paramServerClient)
+    {
+        this.systemConfig = systemConfig;
+        this.paramServerClient = paramServerClient;
+    }
+
+    @Override
+    public String getType()
+    {
+        return "param_set";
+    }
+
+    @Override
+    public Operator newOperator(OperatorContext context)
+    {
+        return new ParamSetOperator(context, paramServerClient);
+    }
+
+    private class ParamSetOperator
+            extends BaseOperator
+    {
+        private ParamServerClient paramServerClient;
+
+        public ParamSetOperator(OperatorContext context, ParamServerClient paramServerClient)
+        {
+            super(context);
+            this.paramServerClient = paramServerClient;
+        }
+
+        @Override
+        public TaskResult runTask()
+        {
+            Optional<String> paramServerType = systemConfig.getOptional("param_server.database.type", String.class);
+            if(!paramServerType.isPresent()){
+                throw new ConfigException("param_server.database.type is required to use this operator.");
+            }
+            Config params = request.getLocalConfig();
+
+            List<String> keys = params.getKeys();
+            if (keys.size() == 0) {
+                throw new ConfigException("no key is set.");
+            }
+
+            paramServerClient.begin();
+
+            for (String key : keys) {
+                Optional<String> value = params.getOptional(key, String.class);
+                if (value.isPresent()) {
+                    paramServerClient.set(key, value.get(), request.getSiteId());
+                }
+            }
+
+            paramServerClient.finalize();
```

Comment:
I think it is necessary.

There are two reasons why I split the transaction handler into `begin()` and `finalize()` method 

1st one is to close connection.

2nd one is, In order to group all data processes in one task into one transaction.
For example, if user about to set several records in one task,  the process of `param_set` must be resulted in all record insert/update is succeeded or all failed. 

So I designed so that Developer can use transaction explicitly, but now I think lambda style is safer because   .. 🤔 

```java
paramServerClient.transaction((client)->{
    for (String key : keys) {
        Optional<String> value = params.getOptional(key, String.class);
        if (value.isPresent()) {
            paramServerClient.set(key, value.get(), request.getSiteId());
        }
    }
});
```

```java
@Override
public void transaction(Consumer<ParamServerClient> process)
{
    begin();
    process.accept(this);
    finalize();
}
```

Reply:
@serihiro The above design looks good to me. About the method name `transaction`, users sometimes could confuse it to understand the behavior of these operators. but, I don't have any preferable names instead of `transaction`. 

Please continue to develop the feature.